### ISSUE TITLE
Premiumizerinstaller.exe failed during reinstallation attempts

### ISSUE THREAD

**Issue by stefanmsanders** [piejanssens/premiumizer#282]
After using the "Uninstall a program" option on my Windows 10 machine to remove Premiumize because it stop running and would not start up again, I attempted to reinstall it by using the Premiumizerinstaller.exe, which is how I installed it the first time without any issues. This time the following errors came up and if I could get some help to fix it I would appreciate it.

unable to execute file: C:\Premiumize\Python\Python.exe
unable to execute file: C:\Premiumize\Python\Scripts\easy_install.exe
unable to execute file: C:\Premiumize\Python\Scripts\pip3.exe

**Comment by neox387** [comment]
Yes it's probably cause python and or git failed to install, would need to remove those before uninstalling premiumizer.

So remove git from apps it might just error with not found, change/repair python then uninstall python

Then try to install premiumizer again =D

**Comment by stefanmsanders** [comment]
Thanks so much for the info.

So after doing that, I'm now getting the following errors:

 

Unable to execute file: C:\Premiumizer\Git\cmd\git.exe

Unable to execute file: C:\Premiumizer\Python\Python.exe

Unable to execute file: C:\Premiumizer\Python\Scripts\easy_install.exe

Unable to execute file: C:\Premiumizer\Python\Scripts\pip3.exe

**Comment by stefanmsanders** [comment]
I just ran it a second time and got the following:

 

 

Unable to execute file: C:\Premiumizer\Python\Python.exe

Unable to execute file: C:\Premiumizer\Python\Scripts\easy_install.exe

Unable to execute file: C:\Premiumizer\Python\Scripts\pip3.exe

**Comment by neox387** [comment]
You could try to download and manually install python 3.8.7 to C:\Premiumizer\Python

**Comment by neox387** [comment]
But if git failed the first time to install premiumizer itself isn't gonna be installed probably..

So again like first post remove git and python then remove premiumizer should work can't do anything else about it :)

**Comment by stefanmsanders** [comment]
Sorry for the confusion, the first time it did work correctly, it ran fine
for about a week, then it stop running. I couldn't get it to restart under
the Windows 10 "Services" option, so I tried uninstalling and reinstalling
it, that's when the problem started. 

 

I created the Python directory you mentioned, then ran the install again. I
think that did it, at least it's installed,  but it's back to the initial
problem.. Permutizer not starting up. Is there a log file somewhere I can
grab to try and figure out why it won't start anymore? I looked in the Setup
log and didn't see a problem there?

 

Btw, thanks for all of the help.

**Comment by xxcriticxx** [comment]
related to #279 

I got around the error by clicking "OK" a couple of times.

**Comment by neox387** [comment]
Log folder in premiumizer folder should say what's wrong but if python or git is not installed it won't run the commands like you can see in the readme to git pull en install requirements

**Comment by stefanmsanders** [comment]
Both (Python & Git) are now installed. Here is a copy of the setup log… but I’m not seeing the log file you mentioned under Premiumizer. Can you tell me spectifically where to go under Premiumizer to see the log file unless you are referring to the Setup log I attached.

 

Thanks

**Comment by neox387** [comment]
I've just updated the installer again to fix these git/python problems in the future.

not sure but I think  Process exit code: 128 means it did not work.

just removed prem / git / python and reinstalled again with new version without problems.
https://github.com/neox387/PremiumizerInstaller/releases/tag/v1.2

**Comment by xxcriticxx** [comment]
@neox387 errors with ver 1.2 installer 

![Capture](https://user-images.githubusercontent.com/15007183/104096175-e65f1700-5268-11eb-9f37-18d9463b7365.PNG)
![Capture2](https://user-images.githubusercontent.com/15007183/104096177-e8c17100-5268-11eb-9bfc-c5d7082268b2.PNG)
![Capture3](https://user-images.githubusercontent.com/15007183/104096178-ebbc6180-5268-11eb-807f-119f3a0f5166.PNG)

**Comment by stefanmsanders** [comment]
yes, seeing issues as well. I notice that the two folders "Git" and "Python" are not being created under Premiumizer, so if fails for me.

**Comment by neox387** [comment]
try to install manually, you will probably also see errors, your windows still thinks they are installed or something and that's why its failing.. ccleaner might help don't know

or its some permission error, like I said no issue on my windows =)

https://www.python.org/ftp/python/3.8.7/python-3.8.7-amd64.exe
https://github.com/git-for-windows/git/releases/download/v2.30.0.windows.1/Git-2.30.0-64-bit.exe

**Comment by stefanmsanders** [comment]
I was able to get it to work by creating the Premiumizer folder, then creating the two subfolders "Git" and "Python" afterwards running the Installer..... thank you so much for the help!!!!!!

**Comment by xxcriticxx** [comment]
ok let me try manual installation

**Comment by neox387** [comment]
> I was able to get it to work by creating the Premiumizer folder, then creating the two subfolders "Git" and "Python" afterwards running the Installer..... thank you so much for the help!!!!!!

Mkay so permission nonsense by windows then maybe running setup as administrator might have worked also i have UAC disabled and also controlled folder access under ransomware protection that thing breaks alot of programs

**Comment by stefanmsanders** [comment]
I tried running it as admin....no luck

**Comment by xxcriticxx** [comment]
what i do in this case?

```
c:\>pip.exe install virtualenv
'pip.exe' is not recognized as an internal or external command,
operable program or batch file.
```

**Comment by stefanmsanders** [comment]
hey xxcriticxx, if you add the full path to pip.exe, then it will work.

**Comment by xxcriticxx** [comment]
now i am stuck on this one: 

env\Scripts\activate.bat

**Comment by stefanmsanders** [comment]
I think it worked for me by doing the same thing under the Scripts directory holding "activate".

**Comment by neox387** [comment]
You could skip the whole virtual env actually it's just some isolation

**Comment by xxcriticxx** [comment]
i think its time to give up on this: 

C:\Users\rocketfish>C:\Users\rocketfish\AppData\Local\Programs\Python\Python39\Scripts\pip install -r requirements.txt
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements.txt'
WARNING: You are using pip version 20.2.3; however, version 20.3.3 is available.
You should consider upgrading via the 'c:\users\rocketfish\appdata\local\programs\python\python39\python.exe -m pip install --upgrade pip' command.

**Comment by xxcriticxx** [comment]
yes i cant start the service

**Comment by stefanmsanders** [comment]
oh, ok.

**Comment by xxcriticxx** [comment]
i made mess on my system congratulation to me :)

**Comment by neox387** [comment]
gonna update the readme =D

git clone https://github.com/piejanssens/premiumizer.git c:\Premiumizer
python -m pip install --upgrade pip
python -m pip install -r c:\Premiumizer\requirements.txt
python C:\Premiumizer\premiumizer\premiumizer.py

**Comment by xxcriticxx** [comment]
first line is no already no good:

c:\>git clone https://github.com/piejanssens/premiumizer.git C:\Premiumizer
'git' is not recognized as an internal or external command,
operable program or batch file.

**Comment by neox387** [comment]
in the git setup the default is now git from command line and also 3th party

so just doing git in cmd should work else your git install is no good =D

**Comment by xxcriticxx** [comment]
fixed git now my python is missing:

c:\>python -m pip install --upgrade pip
Python was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases.

**Comment by neox387** [comment]
Use the checkbox add to path on install or modify and check the box

**Comment by xxcriticxx** [comment]
i check the box and still now working

**Comment by barnabyjones** [comment]
@xxcriticxx did you ever get your install working? I've had similar issues to you. I have got the service installed and running, but I'm still getting the "Downloads Disabled" message. I've been through the logs - both regular and debug - can cannot see any messages relating to downloads being disabled. I have tried setting downloads to both the network location that I have always used and also tried the local disk - but it doesn't make any difference unfortunately.

**Comment by xxcriticxx** [comment]
@barnabyjones no, i gave up on this while ago. What I tell you is to turn off Windows defender while you install your python.  @neox387 suggested using cc cleaner to clean out all directories in windows.

**Comment by wilki24** [comment]
I ran into this issue, and I figured out what's going on (at least in my case).

If you already have this version of python installed for whatever reason, when the Premiumizer installer tries to install here: C:\Premiumizer\Python, it quietly fails to do so. If you manually run the python installer yourself, you'll see that it detects the existing installation of python and offers to repair or remove it, but not install it again somewhere else.

So, C:\Premiumizer\Python never gets installed, and as the Premiumizer installer continues on, it tries to make calls to python, using the hard-coded relative path, which produces an error since python isn't installed there. It looks like this in the installer error log:
2021-03-22 10:14:51.644   -- Run entry --
2021-03-22 10:14:51.644   Run as: Current user
2021-03-22 10:14:51.644   Type: Exec
2021-03-22 10:14:51.644   Filename: C:\Premiumizer\Python\python.exe
2021-03-22 10:14:51.644   Parameters: -m pip install --upgrade pip
2021-03-22 10:14:51.645   Exception message:
2021-03-22 10:14:51.645   Message box (OK):
                          Unable to execute file:
                          C:\Premiumizer\Python\python.exe
                          
                          CreateProcess failed; code 2.
                          The system cannot find the file specified.
                          
Now, I'm not sure how to solve this so that it works both for people who don't have python already installed, and for those that do. Perhaps look into PyInstaller?
